atlasdb-workload-server/src/test/java/com/palantir/atlasdb/workload/invariant/SerializableInvariantTest.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    @Test
+    public void catchesWriteSkew() {
+        AtomicReference<List<InvalidWitnessedTransaction>> invalidTransactions = new AtomicReference<>();
+        List<WitnessedTransaction> transactions = new WitnessedTransactionsBuilder("table")
+                .startTransaction()
+                .write(5, 10, 15)
+                .write(6, 10, 15)
+                .endTransaction()
+                .startTransaction()
+                .read(5, 10, 15)
+                .read(6, 10, 15)
+                .write(5, 10, 0)
+                .endTransaction()
+                .startTransaction()
+                .read(5, 10, 15)
+                .read(6, 10, 15)
+                .write(6, 10, 0)
+                .endTransaction()
+                .build();
+        WorkflowHistory workflowHistory = ImmutableWorkflowHistory.builder()
+                .history(transactions)
+                .transactionStore(readableTransactionStore)
+                .build();
+        SerializableInvariant.INSTANCE.accept(workflowHistory, invalidTransactions::set);
+        assertThat(invalidTransactions).hasValueSatisfying(invalidWitnessedTransactions -> {
+            InvalidWitnessedTransaction invalidWitnessedTransaction =
+                    Iterables.getOnlyElement(invalidWitnessedTransactions);
+            assertThat(invalidWitnessedTransaction.transaction()).isEqualTo(Iterables.getLast(transactions));
+
+            InvalidWitnessedTransactionAction invalidWitnessedTransactionAction =
+                    Iterables.getOnlyElement(invalidWitnessedTransaction.invalidActions());
+            assertThat(invalidWitnessedTransactionAction.mismatchedValue())
+                    .isEqualTo(MismatchedValue.of(Optional.of(15), Optional.of(0)));
+        });
+    }
+
+    @Test
+    public void handlesDeletes() {
+        AtomicReference<List<InvalidWitnessedTransaction>> invalidTransactions = new AtomicReference<>();
+        List<WitnessedTransaction> transactions = new WitnessedTransactionsBuilder("table")
+                .startTransaction()
+                .write(5, 10, 15)
+                .write(6, 10, 15)
+                .endTransaction()
+                .startTransaction()
+                .read(5, 10, 15)
+                .delete(5, 10)
+                .read(5, 10)
+                .endTransaction()
+                .startTransaction()
+                .read(5, 10)
+                .endTransaction()
+                .build();
+        WorkflowHistory workflowHistory = ImmutableWorkflowHistory.builder()
+                .history(transactions)
+                .transactionStore(readableTransactionStore)
+                .build();
+        SerializableInvariant.INSTANCE.accept(workflowHistory, invalidTransactions::set);
+        assertThat(invalidTransactions)
+                .hasValueSatisfying(invalidWitnessedTransactions ->
+                        assertThat(invalidWitnessedTransactions).isEmpty());
```

Comment:
nit: if we decide not to follow the direct List suggestion above, we should be consistent at least (i.e., either asserting on `invalidTransactions.get()` as in L58, or this)